### PR TITLE
Studio support in the client 

### PR DESCRIPTION
**Pull request by Alexis-Jacob**

### CODE REVIEW

File: deepoctl/cmds/studio_helpers/image.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        else:
+            raise RuntimeError("Can't find the dataset {}".format(dataset_name))
+
+        data = {
+            "task": "import_images",
+            "org_slug": org_slug,
+            "cuuid": commit_pk,
+            "dataset_name": dataset_name
+        }
+        if type(path) is not list:
```
Comment (maingoh) [inline deepoctl/cmds/studio_helpers/image.py]: isinstance ?

File: deepoctl/cmds/studio_helpers/task.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+class Task(object):
+    def __init__(self, helper, pk=None, files=None):
+        self._helper = helper
+
+    def retrieve(self, task_id, wait=True):
+        print(task_id)
+        ret = self._helper.get('manage/tasks/{}/'.format(task_id))
+        if not wait:
+            return ret
+        while ret['next'] != None and ret['status'] != 'SUCCESS':
```
Comment (maingoh) [inline deepoctl/cmds/studio_helpers/task.py]: Ideally you should use exp backoff and jiter https://github.com/Deepomatic/deepomatic-client-python/blob/ef579ac78bae3fae00c196bc2fae59876c06a29b/deepomatic/resources/task.py#L65 or the API might suffer

File: deepoctl/cmds/studio_helpers/image.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+            path = [path]
+        if type(path) is list:
+            files = []
+            for elem in path:
+                for file in get_files(elem, recursive):
+                    files.append(('files', open(file, 'rb')))
```
Comment (maingoh) [inline deepoctl/cmds/studio_helpers/image.py]: it's better to close the files explicitely

File: deepoctl/cmds/studio_helpers/http_helper.py [hunk 4]
```diff
@@ -0,0 +1,200 @@
+# -*- coding: utf-8 -*-
+import os
+import sys
+import requests
+import platform
+import json
+from six import string_types
+from requests.structures import CaseInsensitiveDict
+
+class HTTPHelper(object):
```
Comment (maingoh) [inline deepoctl/cmds/studio_helpers/http_helper.py]: We should find a way to use the same as deepomatic-client-python (which I updated this week)

File: deepoctl/cmds/studio_helpers/http_helper.py [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+
+    def make_request(self, func, resource, params=None, data=None, content_type='application/json', files=None, stream=False, *args, **kwargs):
+        if isinstance(data, dict) or isinstance(data, list):
+            if content_type is not None:
+                if content_type.strip() == 'application/json':
+                    data = json.dumps(data)
+                elif content_type.strip() == 'multipart/mixed':
+                    content_type = None  # will be automatically set to multipart
+                    data, files = self.dump_json_for_multipart(data, files)
+                elif content_type.strip() == 'multipart/form':
```
Comment (maingoh) [inline deepoctl/cmds/studio_helpers/http_helper.py]: `multipart/mixed` should be enough no ?

File: deepoctl/cmds/studio_helpers/image.py [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+def get_files(path, recursive=True):
+    if os.path.isfile(path):
+        if path.split('.')[-1].lower() in supported_image_formats:
+            yield path
+    elif os.path.isdir(path) and recursive:
+        for root, dirs, files in os.walk(path, topdown=False):
+            for name in files:
+                if name.split('.')[-1].lower() in supported_image_formats:
+                    yield os.path.join(root, name)
+    elif os.path.isdir(path) and not recursive:
```
Comment (maingoh) [inline deepoctl/cmds/studio_helpers/image.py]: you could have one isdir if

File: deepoctl/cmds/studio_helpers/http_helper.py (hunk 5 above)
Comment (Alexis-Jacob) [reply deepoctl/cmds/studio_helpers/http_helper.py]: no it's not working when I tried that

File: deepoctl/cmds/studio_helpers/image.py (hunk 6 above)
Comment (Alexis-Jacob) [reply deepoctl/cmds/studio_helpers/image.py]: true

File: deepoctl/cmds/studio_helpers/http_helper.py (hunk 4 above)
Comment (Alexis-Jacob) [reply deepoctl/cmds/studio_helpers/http_helper.py]: in another pr no ?

File: deepoctl/cmds/studio_helpers/image.py (hunk 3 above)
Comment (Alexis-Jacob) [reply deepoctl/cmds/studio_helpers/image.py]: it was done in the http_helper but I agree, maybe change that when `studio_helpers` is merged with `deepomatic-client` ?

Comment (maingoh) [reply deepoctl/cmds/studio_helpers/image.py]: Yeah it's not a big deal, just things we should think about later

File: deepoctl/cmds/studio_helpers/http_helper.py (hunk 4 above)
Comment (maingoh) [reply deepoctl/cmds/studio_helpers/http_helper.py]: Yes

File: deepoctl/cmds/studio_helpers/http_helper.py (hunk 5 above)
Comment (maingoh) [reply deepoctl/cmds/studio_helpers/http_helper.py]: Hmm, weird, we will upgrade the http helper from deepomatic-client-python and try again. It should work. Or probably there is a different behavior on studio side.

**Pull request merged**